### PR TITLE
fix(v2): allow using classic theme/preset without the docs plugin

### PR DESCRIPTION
**Pull request by slorber**
## Motivation

Attempt to fix the unability to use the classic theme/preset without the docs plugin 
See https://github.com/facebook/docusaurus/issues/3360

### Have you read the [Contributing Guidelines on pull requests](https://github.com/facebook/docusaurus/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

## Test Plan

`/blog-only/` deploy preview

### CODE REVIEW

**Comment by slorber** [comment]
Hey @SamChou19815 , this is an attempt to fix https://github.com/facebook/docusaurus/issues/3360

I also added a setup that looks like yours in our deploy previews so that we don't regress on this usecase: https://deploy-preview-3382--docusaurus-2.netlify.app/blog-only/


For classic preset without docs, I just allowed to pass `docs: false` in config, which probably make more sense in your case than loading uselessly the docs plugin, as you don't want to use it anyway.

I didn't find a good solution for the useDocs hook so for now I also added a dummy file. 

Currently, it seems the easy fix has 2 choices:
- classic preset loads the docs plugin, but allow to have no docs folder
- classic preset does not load the docs plugin, but we have to fallback in classic theme with the annoying useDocs file.

Do you think one solution is better than the other?
It seems before my refactor we had solution 1, and this PR enables solution 2.
Should we keep using solution 1?

**Comment by SamChou19815** [comment]
> Hey @SamChou19815 , this is an attempt to fix #3360
> 
> I also added a setup that looks like yours in our deploy previews so that we don't regress on this usecase: https://deploy-preview-3382--docusaurus-2.netlify.app/blog-only/
> 
> For classic preset without docs, I just allowed to pass `docs: false` in config, which probably make more sense in your case than loading uselessly the docs plugin, as you don't want to use it anyway.
> 
> I didn't find a good solution for the useDocs hook so for now I also added a dummy file.
> 
> Currently, it seems the easy fix has 2 choices:
> 
> * classic preset loads the docs plugin, but allow to have no docs folder
> * classic preset does not load the docs plugin, but we have to fallback in classic theme with the annoying useDocs file.
> 
> Do you think one solution is better than the other?
> It seems before my refactor we had solution 1, and this PR enables solution 2.
> Should we keep using solution 1?
> 
> At the same time, solution 2 would allow to use classic theme + blog plugin (like you tried) without using the classic preset, so it looks more flexible to me and allow better composition patterns

I think solution 2 makes more sense. Solution 1 will still suffer from the issue of emitting missing docs folder warning. At least for me, I stayed on preset-classic with docs plugin enabled blog-only mode because it requires less config. The new solutions seem to only require one line of additional config and I think that's fine.

File: packages/docusaurus-theme-classic/src/theme/hooks/useDocs.ts [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+ */
+
+// See https://github.com/facebook/docusaurus/issues/3360
+// TODO find a better solution, this shouldn't be needed
+export {};
+/*
+throw new Error(
+  "The docs plugin is not used, so you can't require the useDocs hooks. ",
+);
+ */
```
Comment (SamChou19815) [inline packages/docusaurus-theme-classic/src/theme/hooks/useDocs.ts:15]: We seem to assume here that this dummy theme-classic file is the fallback, and the real hooks in plugin-docs will shadow it. I wonder whether this order of precedence is always desirable. I could imagine another potential use case that requires the opposite order of precedence: plugin-debug defines some default components, and let theme-classic or theme-bootstrap to provide better styling. Although I think the plugin-debug use case is probably less important.

Comment (slorber) [reply packages/docusaurus-theme-classic/src/theme/hooks/useDocs.ts]: Thanks for your insights :) currently trying to figure this out by using the @theme-init alias and see if I can "forward" real useDocs impl. Not sure how to do this properly, Webpack emits a warning on conditional requires (https://github.com/webpack/webpack/issues/7713#issuecomment-467888437)

Comment (slorber) [reply packages/docusaurus-theme-classic/src/theme/hooks/useDocs.ts]: Seems to work according to the tests but there is an annoying webpack warning, not sure how to improve this.

**Comment by slorber** [comment]
Basically this shows this webpack error on build:

image

https://github.com/webpack/webpack/issues/7713#issuecomment-467888437

I was actually able to  remove that warning using warningsFilter webpack option (not sure why but we seem to reimplement it in Docusaurus btw): https://webpack.js.org/configuration/stats/#statswarningsfilter

**Comment by slorber** [comment]
closing as CLA bot seems to not perform the check :(

**Pull request merged**